src/lein_monolith/target.clj

```diff
@@ -0,0 +1,85 @@
+(ns lein-monolith.target
+  "Functions for constructing and operating on dependency closures."
+  (:require
+    [clojure.set :as set]
+    [clojure.string :as str]
+    (lein-monolith
+      [config :as config]
+      [dependency :as dep])))
+
+
+(def selection-opts
+  {:in* 1
+   :upstream-of* 1
+   :downstream-of* 1
+   :skip* 1
+   :select* 1})
+
+
+(defn- resolve-projects
+  "Returns a set of project names that have been resolved from the given
+  sequence, which may consist of multiple comma-separated lists. Ignores names
+  which do not map to a project."
+  [subprojects name-args]
+  (some->>
+    (seq name-args)
+    (mapcat #(str/split % #"\s*,\s*"))
+    (map read-string)
+    (keep (partial dep/resolve-name (keys subprojects)))
+    (set)))
+
+
+(defn- resolve-selectors
+  "Returns a selection function to filter projects based on the `:select`
+  options passed. Multiple selection functions apply cumulative layers of
+  filtering, meaning a project must pass _every_ selector to be included.
+  Returns nil if no selectors were specified."
+  [monolith select-args]
+  (some->>
+    (seq select-args)
+    (map read-string)
```

Comment:
`read-string`? scandalous! I mean, I guess the worst thing the user can do is sabotage their own JVM...